### ISSUE TITLE
NodeByUri will not query WooCommerce products since v1.8.1

### ISSUE THREAD

**Issue by jmotes-lapero** [wp-graphql/wp-graphql#2394]
v1.8.2:

<img width="946" alt="image" src="https://user-images.githubusercontent.com/94563084/170400530-596198d8-d881-4933-8b3a-cb29e6cab311.png">

v1.8.0:

<img width="1072" alt="image" src="https://user-images.githubusercontent.com/94563084/170400578-c2634191-6dd8-4d84-a1b1-11275c2660cb.png">

I'm using WooGraphQL v0.11.0 but experienced the issue with v0.10.5 as well.

**Comment by jasonbahl** [comment]
@jmotes-lapero Can you provide a bit more info to reproduce this? 

Like, are you able to reproduce this on a minimal install of WordPress with just WPGraphQL, WooCommerce and WooGraphQL and nothing else? 

If so, what data set is needed to reproduce this?

Is the product published? etc

**Comment by jasonbahl** [comment]
@jmotes-lapero does it work in v1.8.1 as well? Or is that version broken too? Would be good to narrow down the culprit to 1.8.1 or 1.8.2 if possible.

**Comment by jmotes-lapero** [comment]
@jasonbahl it breaks starting with 1.8.1 in our project. I'll work on a minimal reproduction today. I assumed it would be easy to reproduce since 1.8.0 works fine for us, but sounds like I was wrong. Sorry about that!

**Comment by jmotes-lapero** [comment]
Rolling back to 1.8.0:

<img width="1306" alt="image" src="https://user-images.githubusercontent.com/94563084/170510127-11ae3083-30ec-4bdf-b7b1-0e343dd1531f.png">

And on 1.8.1:

<img width="1108" alt="image" src="https://user-images.githubusercontent.com/94563084/170510337-68f5dd1f-2833-42e9-8c8f-88f4a4390778.png">

**Comment by justlevine** [comment]
@jmotes-lapero Are you sure the only thing that changed is the WPGraphQL version and not maybe the WC version? Because I *cant* get it to work in v1.8.0, v1.7.2 or v1.6.12.

**v1.8.0**
![image](https://user-images.githubusercontent.com/29322304/170587567-5189e7cd-5a7d-42d4-9844-a1f5ad83cbb3.png)

**v1.6.12**
![image](https://user-images.githubusercontent.com/29322304/170588413-3fdc6941-a06a-4fb6-83eb-1ba87566571f.png)

**Comment by justlevine** [comment]
Okay, seems like this is an issue with wp-graphql-woocommerce's [`Core_Schema_Filters::skip_type_registry()`](https://github.com/wp-graphql/wp-graphql-woocommerce/blob/adf9c42001ec53d49cc5a107a940d1ea0700ae90/includes/class-core-schema-filters.php#L172), which removes the `product` CPT from `WPGraphQL::get_allowed_post_types()`. 

This was also a problem pre #2353 (where products were missing some autogenerated Interfaces/Fields/Connections and left out of some Unions ), but since #2356 started using `WPGraphQL::get_allowed_post_types()` consistently throughout the plugin, it's now a bigger issue than previously.

Long term, I'd assume the ideal solution is to dust off #1486, but I'm still trying to grok how wp-graphql-woocommerce handles the `product` interface and the correlating types to figure out a short-term workaround.

@kidunot89 - any ideas?

**Comment by justlevine** [comment]
> I think if we did something similar to #1486, I'd like to at least nest the config under a graphql field.

I like the idea of this, but not sure how nicely it would play with the underlying [`wp_filter_object_list()`](https://developer.wordpress.org/reference/functions/wp_filter_object_list/), and being able to easily filter allowed types by individual properties is essential to letting us conditionally register what we need to.

We could swap out that function for custom logic if we don't mind breaking away from the pattern used by WP core, but it does add complexity.

Thoughts?

I'm still not not seeing a clear path forward to resolve _this_ issue with the current WPGraphQL codebase, so imo it makes sense to lean into whatever parts of #1486 we can (breaking it up so we can ship a fix sooner), instead of additional logic / new filters that a refreshed #1486 would make obsolete.

**Comment by jasonbahl** [comment]
@justlevine good point about the nesting. Perhaps we just use prefixing then. 

```php
register_post_type( 'post_type_name', [ 
    'show_in_graphql' => true,
    'graphql_single_name' => 'MyType',
    'graphql_plural_name' => 'MyTypes',

    // The "Kind" of GraphQL Type to register. This would allow the `MyType` type to be registered as an object type (default) or Interface type. WooCommerce products would want to be an Interface.
    'graphql_kind' => 'object' / 'interface', 

    // function used to resolve the node type. (see: https://github.com/jacobarriola/woographql-product-bundles/issues/5, https://github.com/sorensenjg/woographql-composite-products/issues/2)
    'graphql_resolve_type' => function( $node ) { ... },

    // register interfaces to the type. This could either be additive (my preference) to the default interfaces (such as `Node`, etc), or override the defaults. For example, if you pass `MyCustomInterface` it could be applied _in addition to_ the default interfaces such as `Node`, `DatabaseIdentifier`, etc, or it could override and be the _only_ Interface unless the others were manually added here as well. 🤔
    'graphql_interfaces' => [], 

    // whether the field should register a root field/connection
    'graphql_register_root_field' => true/false,
    'graphql_register_root_connection' => true/false,
]);
```

**Comment by jasonbahl** [comment]
I think some of the naming of arguments in #1486 aren't declarative enough. 

like if I wasn't seasoned in WPGraphQL and saw a post type in a codebase like so:

```php
register_post_type( 'my_new_type', [
  ...
  'connect_to_taxonomies' => true/false
]);
```

I wouldn't know what that meant.

But, something more explicit like: 

``````php
register_post_type( 'my_new_type', [
  ...
  'graphql_register_taxonomy_connections' => true/false
]);
```

would help me understand the impact of this field in the registration of the post type.

**Comment by jmotes-lapero** [comment]
@jasonbahl this bug is preventing us from upgrading to v1.12 so that we can make use of WP Smart Cache. Is there anything I can do to help? 

#1486 that's mentioned above has been superceded by #2399 and merged already, but this is still an issue in v1.12.0.

**Comment by justlevine** [comment]
Working in WPGraphQL v1.13.7 + WPGraphQL WooCommerce v0.12.0

![image](https://user-images.githubusercontent.com/29322304/208267862-d73ad5b4-62d0-4b42-9ae9-831db0e169f0.png)
